packages/context/src/__tests__/unit/handler-chain.unit.ts

```diff
@@ -0,0 +1,168 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  compareBindingsByTag,
+  Context,
+  filterByTag,
+  Handler,
+  InvocationHandlerChain,
+  Next,
+} from '../..';
+
+describe('InvocationHandlerChain', () => {
+  let ctx: Context;
+  let handlerChain: InvocationHandlerChain;
+  let events: string[];
+
+  beforeEach(givenContext);
+
+  it('invokes handler functions', async () => {
+    givenHandlerChain(
+      givenNamedHandler('handler1'),
+      givenNamedHandler('handler2'),
+    );
+    const result = await handlerChain.invokeHandlers();
+    expect(events).to.eql([
+      'before-handler1',
+      'before-handler2',
+      'after-handler2',
+      'after-handler1',
+    ]);
+    expect(result).to.be.undefined();
+  });
+
+  it('honors return value', async () => {
+    givenHandlerChain(givenNamedHandler('handler1'), async (context, next) => {
+      await next();
+      return 'ABC';
+    });
+    const result = await handlerChain.invokeHandlers();
+    expect(result).to.eql('ABC');
+  });
+
+  it('skips downstream handlers if next is not invoked', async () => {
+    givenHandlerChain(async (context, next) => {
+      return 'ABC';
+    }, givenNamedHandler('handler1'));
```

Comment:
Since the named handler is second in the chain, let's use `handler2` as the name.